client/state/lib/middleware.js

```diff
@@ -211,7 +210,6 @@ const handler = ( dispatch, action, getState ) => {
 
 		case SITE_RECEIVE:
 		case SITES_RECEIVE:
```

Comment:
One thing we forgot about in #20747: here we should watch for `InvitesActionTypes.RECEIVE_INVITE_ACCEPTED_SUCCESS`, too: it changes the site list.

Or consider dispatching `SITES_RECEIVE` separately when invite is accepted -- it starts to look like a really good idea now.

Reply:
That's a very good catch, Jarda!

I went with the second approach in a81cc7a and now clicking on the "accept invite" works properly: the user is shown all the sidebar items instead of seeing a buggy view: http://cld.wthms.co/aSCGfX